front/src/App.tsx

```diff
@@ -17,17 +17,29 @@ type UnvalidatedSpell = {
 
 function App() {
     const [spells, setSpells] = useState<Spell[]>([]);
+    const [filteredList, setFilteredList] = useState<Spell[]>(spells);
+    const [characterName, setCharacterName] = useState<string>("Josh Mann");
+
+    const handleSearch = (query: string) => {
+        if (query.trim() == "") {
+            setFilteredList(spells);
+        } else {
+            const lowerCaseQuery = query.trim().toLowerCase();
+            setFilteredList(
+                spells.filter((spell) =>
+                    spell.name.toLowerCase().includes(lowerCaseQuery)
+                )
+            );
+        }
```

Comment:
```suggestion
        const query = query.trim().toLowerCase()
        
        if (query === "") {
            setFilteredList(spells);
            return
        }
        
        setFilteredList(
            spells.filter((spell) =>
                spell.name.toLowerCase().includes(query)
            )
        );
```
We should always use `===`. It's just not worth the headache of dealing with `==`. Also, guard clauses helping to reduce the nesting of code, making it easier to read.